src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+      switch (action.type) {
+        case ('addProperties'): {
+          Object.assign(state, action.extraData);
+          break;
+        }
+
+        case ('removeProperties'): {
+          for (const props of action.keysToRemove) {
+            delete state[props];
+          }
+          break;
+        }
+
+        case ('clear'): {
+          for (const keys in state) {
+            delete state[keys];
+          }
+        }
+      }
+    }
```

Comment:
Use 2 spaces indentation, not tabs

Reply:
@illiarerun thank you for that, would be impossible to become a programmer without that stuff.

Reply:
))